File: BE-Sieuthidienmay/src/main/java/com/sapo/mock/techshop/entity/Collection.java
package com.sapo.mock.techshop.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@ToString
@Table(name = "collections")
public class Collection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    @JsonBackReference // will not be serialized to JSON
    private Category category;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "manufacturer_id")
    @JsonBackReference // will not be serialized to JSON
    private Manufacturer manufacturer;

    @OneToMany(mappedBy = "collection", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JsonManagedReference
    List<Product> products;

    @CreationTimestamp
    private Date created_at;
    @UpdateTimestamp
    private Date updated_at;
}

File: FE_Sieuthidienmay/src/types/collection.ts
import { Category } from "./category";
import { Manufacturer } from "./manufacturer";

export interface Collection{
    id: number;
    name: string;
    category: Category;
    manufacturer: Manufacturer;
    products: Product[];
    created_at: string;
    updated_at: string;
  }

  export interface CollectionRequest{
    name: string;
    category_id: number;
    manufacturer_id: number;
  }

  


export interface Product{
  id: number;
  name: string;
  importingPrice: number;
  sellingPrice: number;
  discount: number;
  remainingAmount:number;
  soldAmount:number;
  detail:string;
  rating: number;
  imgLink: string;
  createdAt: string;
}